### PR TITLE
feat: add week/month interval filter on statistics

### PR DESCRIPTION
**Pull request by sborrazas**

### CODE REVIEW

File: lib/ae_mdw/stats.ex [hunk 1]
```diff
@@ -243,23 +251,44 @@ defmodule AeMdw.Stats do
...
     fn direction ->
       Collection.stream(state, Model.Statistic, direction, key_boundary, cursor)
     end
   end
 
+  defp render_statistic(state, {{:transactions, _tx_tag}, :month, interval_start} = statistic_key) do
+    Model.statistic(count: count) = State.fetch!(state, Model.Statistic, statistic_key)
+
+    %{
+      start_date: months_to_unix(interval_start),
```
Comment (jyeshe) [inline lib/ae_mdw/stats.ex:276]: how about using an iso date instead of the timestamp? It's compact, makes it already formatted to render on the frontend and is also human readable.

**Review by jyeshe — COMMENTED** [review]
Works well, good idea to make the month also 0-indexed. Only for the format `start_date`, would pick the ISO.

File: test/ae_mdw_web/controllers/stats_controller_test.exs [hunk 2]
```diff
@@ -165,4 +166,95 @@ defmodule AeMdwWeb.StatsControllerTest do
...
   end
+
+  test "when interval_by = week, it returns the count of transactions for the latest weekly periods",
+       %{
+         conn: conn,
+         store: store
+       } do
+    start_interval1 = 0
+    start_interval2 = 1_000
+    start_interval3 = 2_000
```
Comment (jyeshe) [inline test/ae_mdw_web/controllers/stats_controller_test.exs:177]: may we use multiples of 52 and add a test for a leap year?

File: lib/ae_mdw/stats.ex (hunk 1 above)
Comment (sborrazas) [reply lib/ae_mdw/stats.ex:276]: this is how the node formats time already, so I'd thought I wouldn't change it to avoid having the client implement two separate decoders for datetimes

File: test/ae_mdw_web/controllers/stats_controller_test.exs (hunk 2 above)
Comment (sborrazas) [reply test/ae_mdw_web/controllers/stats_controller_test.exs:177]: All three day/week/month implementations start counting for days/week/months since epoch, so leap years wouldn't really be an edge-case. I've added a testcase for 1972-1973 for which 1972 is a leap year anyway

File: lib/ae_mdw/stats.ex (hunk 1 above)
Comment (jyeshe) [reply lib/ae_mdw/stats.ex:276]: With ISO date there is no need for decoders, it's yyyy-mm-dd. Even though the issue says `start_time`, the name chosen in the PR `start_date` makes it more precise since there is no time related to this stats.

File: test/ae_mdw_web/controllers/stats_controller_test.exs (hunk 2 above)
Comment (jyeshe) [reply test/ae_mdw_web/controllers/stats_controller_test.exs:177]: With 52 or 53 weeks, it can give a different datetime (timestamp) or date depending on the implementation.
The idea of the test would be not to rely on the implementation and check easily without external computations (on how `1_000` should be represented as date).

**Review by jyeshe — APPROVED** [review]
